utils/filecache/interfaces.go

```diff
@@ -0,0 +1,42 @@
+package filecache
+
+import (
+	"context"
+	"io"
+
+	"github.com/ARM-software/golang-utils/utils/filesystem"
+)
+
+// IFileCache provides a filesystem cache for files and directories.
+// Cache entries are identified by a key and live for a configurable TTL (time-to-live).
+// An internal garbage collector periodically evicts expired entries; the period is also configurable.
+type IFileCache interface {
+	io.Closer
+
+	// Calls the entry provider to store an entry into the cache under the provided `key`, and starts the expiration
+	// timer based on the configured TTL.
+	// Returns an error if the path doesn't exist, the key already exists in the cache, the copy process fails or
+	// the cache is closed.
+	Store(ctx context.Context, key string) error
+
+	// Evicts the cache entry identified by `key`, removing the stored cached entry immediately.
+	// Returns an error if the key does not exist in the cache, the deleting process fails or the cache is closed.
```

Comment:
if the key does not exist in the cache, no need to return an error. eviction is in a way successful